app/models/katello/repository.rb

```diff
@@ -92,9 +89,13 @@ class Repository < Katello::Model
     scope :non_puppet, where("content_type != ?", PUPPET_TYPE)
     scope :non_archived, where('environment_id is not NULL')
     scope :archived, where('environment_id is NULL')
-
     scope :completer_scope, lambda { |options| options[table_name].call(self) }
 
+    scoped_search :on => :name, :complete_value => true
+    scoped_search :rename => :product, :on => :name, :through => :product, :complete_value => true
+    scoped_search :on => :content_type, :complete_value => Katello::Repository::TYPES.each_with_object({})  { |value, hash| hash[value.to_sym] = value }
+    scoped_search :on => :content_view_id, :in => :content_view_repositories
```

Comment:
actually this isn't currently possible due to https://github.com/wvanbergen/scoped_search/issues/131